google/resumable_media/_download.py

```diff
@@ -124,8 +124,23 @@ class Download(DownloadBase):
             ``start`` to the end of the media.
         headers (Optional[Mapping[str, str]]): Extra headers that should
             be sent with the request, e.g. headers for encrypted data.
+        checksum Optional([str]: The type of checksum to compute to verify
+            the integrity of the object. The response headers must contain
+            a checksum of the requested type. If the headers lack an
+            appropriate checksum (for instance in the case of transcoded or
+            ranged downloads where the remote service does not know the
+            correct checksum) an INFO-level log will be emitted. Supported
+            values are "md5", "crc32c" and None.
     """
 
+    def __init__(
+        self, media_url, stream=None, start=None, end=None, headers=None, checksum="md5"
```

Comment:
@crwilcox per discussion with @andrewsg, the only gotcha with checksum being set in the constructor is that a user can manipulate the value after initialization and cause a failure during consume. 

I think this may an edge case and not too concerned. Bringing it up as remaining talking point.

Reply:
I addressed this by doing validation in the consume method, not in the constructor. Manipulating the value after initialization will not cause an issue.

Reply:
Sounds good, then that just leaves my other comment on documentation. 

Thanks @andrewsg